### ISSUE TITLE
Stand O' Food (PSP PSN Minis) - Blackscreen issue

### ISSUE THREAD

**Issue by t3rr41n** [hrydgard/ppsspp#14863]
### What happens?
Screen only shows "This game saves data automatically at certain points. Do not remove memory stick..." after confirm only shows blackscreen, but there is sounds.

### What *should* happen?
The game starts showing image after confirming"This game saves data automatically at certain points. Do not remove memory stick..."

#### Which game or games?

Stand O' Food - NPEZ00002

#### Checklist

 - [ ] Test in the [latest git build](https://buildbot.orphis.net/ppsspp/) in case it's already fixed.
 - [X] [Search for other reports](https://github.com/hrydgard/ppsspp/search?q=my+issue&type=issues) of the same issue.
 - [X] Try resetting settings or older versions and include if the issue is related.
 - [ ] Include logs or screenshots of issue.

**Comment by Gamemulatorer** [comment]
Cannot reproduce using the latest git build.
Phone use is Cloudfone Excite Prime Mali-450 GPU.
![Screenshot_2021-09-15-20-43-17](https://user-images.githubusercontent.com/37603562/133435595-813985a6-04a3-4486-a7b4-82a0f8982e72.png)
![Screenshot_2021-09-15-20-43-37](https://user-images.githubusercontent.com/37603562/133435614-db7d98ec-8b55-42a9-acb6-1be34fed4be5.png)

**Comment by t3rr41n** [comment]
Do this is also due to corrupted .iso file ?
Because when i download it's says 9MB download, but when unpacked it was 120MB

**Comment by t3rr41n** [comment]
Can you give me links to download site
When i download the .iso file, i download it from my usual download site
Anyway, do there was a new version of PPSSPP ?

**Comment by Gamemulatorer** [comment]
> Can you give me links to download site
> When i download the .iso file, i download it from my usual download site.

Download link is forbbiden for this site piracy is not allowed by Henryk and Unknown.

> Anyway, do there was a new version of PPSSPP ?

https://buildbot.orphis.net/ppsspp/index.php?m=fulllist

**Comment by ANR2ME** [comment]
Btw, are you using Frameskipping? as i remembered there is a game that shows blackscreen when using frameskipping

**Comment by Panderner** [comment]
> Btw, are you using Frameskipping?
> As i remembered there is a game that shows blackscreen when using frameskipping, not sure whether it have been fixed on the latest dev build or not.
> 
> Edit: probably unrelated

I used frameskip for this game and it's working fine.

**Comment by t3rr41n** [comment]
Already reinstall the PPSSPP, issue still persists
Already download another .cso of the game, issue still persists
Maybe i will try the latest git build

**Comment by Gamemulatorer** [comment]
@t3rr41n can you upload the screen recording of the issue?

**Comment by ResurrectionRedmi9** [comment]
> Already download another .cso of the game, issue still persists.

Do not use cso format.

> Maybe i will try the latest git build

Yes try the ppsspp buildbot.

**Comment by Panderner** [comment]
@t3rr41n what is your device are you using?

**Comment by Gamemulatorer** [comment]
> @t3rr41n what is your device are you using?

He already stated that his phone is Redmi 2 Pro Android Lollipop.

**Comment by Panderner** [comment]
Hmm ancient snapdragon phone. Can you produce GE Dump?

**Comment by Gamemulatorer** [comment]
@t3rr41n no feedback/outdated.

**Comment by Panderner** [comment]
> Do not use cso format.

I played this game using CSO format working fine.

**Comment by unknownbrackets** [comment]
https://report.ppsspp.org/game/NPEZ00002_1.01

Hm, not many reports.  CSO ought to be completely fine for this or any other game (I personally run every game using CSO.)

It's possible this is depth clamp or dual source alpha related.  A GE frame dump would indeed be helpful.  

See here for instructions - it's not hard and works on Android too:
https://github.com/hrydgard/ppsspp/wiki/How-to-create-a-frame-dump

You can zip that and then drag and drop it into a reply here.

-[Unknown]

**Comment by Panderner** [comment]
Here's a GE Dumps:
[Menu.zip](https://github.com/hrydgard/ppsspp/files/7192509/Menu.zip)
[In-game.zip](https://github.com/hrydgard/ppsspp/files/7192510/In-game.zip)

**Comment by t3rr41n** [comment]
Sorry, last week's was crazy busy. Here the GE Dump https://drive.google.com/file/d/1F3J7PdEAMvGEF3DB54oVmbWNyQNqGXaQ/view?usp=drivesdk

**Comment by t3rr41n** [comment]
Here's the screen recording for a clearer idea of the problem
https://drive.google.com/file/d/1FCmOpoJ4YUroK5Mrx6Aaju82BrlMpsWT/view?usp=drivesdk

**Comment by ANR2ME** [comment]
In the past, black screen issue can be solved by extracting the content of the ISO https://www.youtube.com/watch?v=Gjq7jKMMMvE

But based on your video, seems to be flickering issue instead of black screen.
Btw, have you tried changing the Buffered rendering Mode settings?

**Comment by Panderner** [comment]
> In the past, black screen issue can be solved by extracting the content of the ISO https://www.youtube.com/watch?v=Gjq7jKMMMvE

@ANR2ME You don't need to extract the content of the ISO it's working fine.

**Comment by Panderner** [comment]
@t3rr41n Check the CRC if the game is same as a working CRC.

**Comment by t3rr41n** [comment]
> In the past, black screen issue can be solved by extracting the content of the ISO based on this PPSSPP video (not sure whether the issue has been reported in the past or not tho) https://www.youtube.com/watch?v=Gjq7jKMMMvE
> 
> But based on your video, seems to be flickering issue instead of black screen.
> Btw, have you tried changing the Buffered rendering Mode settings?

I use buffered rendering and simulate block transfer. Also the buffered rendering set on natural color, but with both options turned off the issue still persists.

**Comment by t3rr41n** [comment]
> @t3rr41n Check the CRC if the game is same as a working CRC.

How i check the CRC of the game and the real one ?

**Comment by t3rr41n** [comment]
> > In the past, black screen issue can be solved by extracting the content of the ISO https://www.youtube.com/watch?v=Gjq7jKMMMvE
> 
> @ANR2ME You don't need to extract the content of the ISO it's working fine.

Right, when i test the ISO (decrypt) everything fine. No corrupted file, etc.

**Comment by Panderner** [comment]
> > @t3rr41n Check the CRC if the game is same as a working CRC.
> 
> How i check the CRC of the game and the real one ?

Long press the game and then press calculate CRC.

**Comment by ANR2ME** [comment]
> 
> 
> > > In the past, black screen issue can be solved by extracting the content of the ISO https://www.youtube.com/watch?v=Gjq7jKMMMvE
> > 
> > 
> > @ANR2ME You don't need to extract the content of the ISO it's working fine.
> 
> Right, when i test the ISO (decrypt) everything fine. No corrupted file, etc.

Well based on that video on youtube the ISO didn't have any issue being extracted with another app too, but it looked like PPSSPP seems to fails to load some of the files from the ISO (or may be failed to extract it?)

However, that video was an old version (1.9.4) of PPSSPP, not sure whether it has been reported before or already fixed by now tho.

**Comment by Gamemulatorer** [comment]
@ANR2ME thanks for the yt link got the idea and play Resistance Retribution without freeze hahaha 🤣
https://github.com/hrydgard/ppsspp/issues/4626#issuecomment-924642157

**Comment by t3rr41n** [comment]
![Screenshot_2021-09-24-06-42-50-487_org ppsspp  #ppssppgold](https://user-images.githubusercontent.com/82008447/134597735-52d2ece7-b4d9-4435-880f-4611b2aef290.png)

Didn't see the calculate CRC, where is the options ?

> > > @t3rr41n Check the CRC if the game is same as a working CRC.
> > 
> > 
> > How i check the CRC of the game and the real one ?
> 
> @t3rr41n Long press the game and then press calculate CRC.

**Comment by Panderner** [comment]
> Didn't see the calculate CRC, where is the options ?

Scroll down to see calculate CRC.

**Comment by hrydgard** [comment]
> > Didn't see the calculate CRC, where is the options ?
> 
> Scroll down to see calculate CRC.

yeah, the list of options has gotten too long for some screens... I think we might need to add an indicator of some sorts that there are more items

**Comment by t3rr41n** [comment]
On the bottom of the list only "Use UI Background"
Is there any option I need to enable ?

**Comment by Panderner** [comment]
If your phone is too small to see Calculate CRC is not here. You can scroll down like this:

https://user-images.githubusercontent.com/37503397/134665101-b6da6c42-82ed-4f30-8500-fa9734745f21.mp4

**Comment by ResurrectionRedmi9** [comment]
@t3rr41n no feedback/outdated?

**Comment by unknownbrackets** [comment]
> yeah, the list of options has gotten too long for some screens... I think we might need to add an indicator of some sorts that there are more items

Maybe just move it slightly lower?  Usually the best indicator is a half visible item.

-[Unknown]

**Comment by unknownbrackets** [comment]
Well, that's interesting.  This game uses a viewport offset and scale of 0.  I could see that having... interesting effects on some hardware/drivers in old non-accurate depth mode.  This is probably fixed whenever accurate depth is used, though.

-[Unknown]

**Comment by t3rr41n** [comment]
> If your phone is too small to see Calculate CRC is not here. You can scroll down like this:
> 
>  Example.mp4

![Screenshot_2021-09-27-07-40-18-315_org ppsspp ppssppgold](https://user-images.githubusercontent.com/82008447/134830265-69133c63-fecc-421d-8c6a-b110b7ebe990.png)

Here's the Screenshot at the bottom of the list, there were no Calculate CRC option. Is new version is needed ?

**Comment by Panderner** [comment]
Umm yes @t3rr41n please use the latest build.

**Comment by ResurrectionRedmi9** [comment]
@t3rr41n no feedback/outdated!

**Comment by Panderner** [comment]
@ResurrectionRedmi9 do not mention someone for no feedback/outdated please stop.

**Comment by t3rr41n** [comment]
Already use the latest git build, the issue still persists
Already calculate the CRC, it's the exact match as stated on the download site (0CEEA16B)

**Comment by Panderner** [comment]
> 0CEEA16B

Hmm that CRC is working fine for me i tested.

**Comment by t3rr41n** [comment]
So... any suggestions ?

**Comment by Gamemulatorer** [comment]
@t3rr41n use ppsspp v1.12 https://www.ppsspp.org/downloads.html

**Comment by Gamemulatorer** [comment]
> I will try on another device, and see if the issue still persists

Make sure are you using default settings.

**Comment by unknownbrackets** [comment]
Okay, so to recap what we know:

 * It's not working on a Adreno 306 device.
 * It's happening with a good copy of the disc.
 * It's working on older Mali 4xx devices.
 * The most interesting thing it does in rendering seems to be a Z scale/offset of 0, which is pretty uncommon and possibly even a (harmless on the PSP) mistake in the game.

-[Unknown]

**Comment by Gamemulatorer** [comment]
@t3rr41n can still reproduce this issue?

**Comment by t3rr41n** [comment]
Yes, it seems that it was really a Adreno 3xx series issue. Already tried on non-Adreno device it's run properly

**Comment by Gamemulatorer** [comment]
> Here's the screen recording for a clearer idea of the problem https://drive.google.com/file/d/1FCmOpoJ4YUroK5Mrx6Aaju82BrlMpsWT/view?usp=drivesdk

I will try to reproduce this on my vivo y51a adreno 306.

**Comment by Gamemulatorer** [comment]
> > Here's the screen recording for a clearer idea of the problem https://drive.google.com/file/d/1FCmOpoJ4YUroK5Mrx6Aaju82BrlMpsWT/view?usp=drivesdk
> 
> I will try to reproduce this on my vivo y51a adreno 306.

I cannot reproduce this issue of yours @t3rr41n using the latest build 🤔

**Comment by Panderner** [comment]
hmm it seems the vivo y51a and Redmi 2 Pro are using a same GPU.

**Comment by unknownbrackets** [comment]
It's possible the change I made for drawing/clearing recently may have improved this.

-[Unknown]

**Comment by Gamemulatorer** [comment]
t3rr41n no feedback?

**Comment by t3rr41n** [comment]
Okay, i will try it again using the latest build

**Comment by Gamemulatorer** [comment]
> Okay, i will try it again using the latest build

How about now?

**Comment by t3rr41n** [comment]
Looks like the issue was solved in the latest release

**Comment by unknownbrackets** [comment]
Great, thanks for the feedback.  Going to close this then.

-[Unknown]
